ts/image-occlusion/tools/tool-zoom.ts

```diff
@@ -156,26 +156,50 @@ const onDrag = (canvas, opt) => {
     redraw(canvas);
 };
 
-export const onDragX = (canvas: fabric.Canvas, opt: WheelEvent) => {
-    const delta = opt.deltaX;
+export const onWheelDrag = (canvas: fabric.Canvas, event: WheelEvent) => {
+    const deltaX = event.deltaX;
+    const deltaY = event.deltaY;
     const vpt = canvas.viewportTransform;
-    canvas.lastPosX = opt.clientX;
+    canvas.lastPosX = event.clientX;
+    canvas.lastPosY = event.clientY;
 
-    vpt[4] -= delta;
-    canvas.lastPosX -= delta;
+    vpt[4] -= deltaX;
+    vpt[5] -= deltaY;
+
+    canvas.lastPosX -= deltaX;
+    canvas.lastPosY -= deltaY;
     canvas.setViewportTransform(vpt);
     constrainBoundsAroundBgImage(canvas);
     redraw(canvas);
 };
 
-export const onDragY = (canvas: fabric.Canvas, opt: WheelEvent) => {
-    const delta = opt.deltaY;
+// for mac trackpad
+export const onGesture = (event) => {
```

Comment:
I have implemented this solution but currently I do not have access to hardware to test this for trackpad, I request you test it, it new changes required, it may be done in this method.